Tribler/Test/GUI/test_gui.py

```diff
@@ -278,112 +281,63 @@ def test_home_page_channels(self):
+        QTest.keyClick(widget.channel_torrents_filter_input, '1')
+        self.wait_for_list_populated(widget.content_table)
+        self.screenshot(window, name="%s-filtered" % widget_name)
+        self.assertLessEqual(widget.content_table.verticalHeader().count(), old_num_items)
 
         # Unsubscribe and subscribe again
-        index = self.get_index_of_row(window.subscribed_channels_list, 0)
-        window.subscribed_channels_list.on_subscribe_control_clicked(index)
+        index = self.get_index_of_row(widget.content_table, 0)
+        widget.content_table.on_subscribe_control_clicked(index)
         QTest.qWait(200)
-        self.screenshot(window, name="unsubscribed")
-        window.subscribed_channels_list.on_subscribe_control_clicked(index)
+        self.screenshot(window, name="%s-unsubscribed" % widget_name)
+        widget.content_table.on_subscribe_control_clicked(index)
         QTest.qWait(200)
 
-    def test_discovered_page(self):
-        QTest.mouseClick(window.left_menu_button_discovered, Qt.LeftButton)
-        self.wait_for_list_populated(window.discovered_channels_list)
-        self.screenshot(window, name="discovered_page")
-
-        # Sort
-        window.discovered_channels_list.sortByColumn(1, 1)
-        self.wait_for_list_populated(window.discovered_channels_list)
-        self.screenshot(window, name="discovered_sorted")
-        self.assertLessEqual(window.discovered_channels_list.verticalHeader().count(), 50)
-
-        # Filter
-        old_num_items = window.discovered_channels_list.verticalHeader().count()
-        QTest.keyClick(window.discovered_channels_filter_input, '1')
-        self.wait_for_list_populated(window.discovered_channels_list)
-        self.screenshot(window, name="discovered_filtered")
-        self.assertLessEqual(window.discovered_channels_list.verticalHeader().count(), old_num_items)
-
-    def test_channel_torrents(self):
-        QTest.mouseClick(window.left_menu_button_subscriptions, Qt.LeftButton)
-        self.wait_for_list_populated(window.subscribed_channels_list)
-        index = self.get_index_of_row(window.subscribed_channels_list, 0)
-        window.subscribed_channels_list.on_table_item_clicked(index)
-        self.wait_for_list_populated(window.channel_page_container.content_table)
-        self.screenshot(window, name="channel_torrents_loaded")
+        # Test channel view
+        index = self.get_index_of_row(widget.content_table, 0)
+        widget.content_table.on_table_item_clicked(index)
+        self.wait_for_list_populated(widget.content_table)
+        self.screenshot(window, name="%s-channel_loaded" % widget_name)
 
+        # FIXME: credit mining for collections!
         # Toggle credit mining
-        QTest.mouseClick(window.credit_mining_button, Qt.LeftButton)
-        self.wait_for_signal(window.subscription_widget.credit_mining_toggled)
+        # QTest.mouseClick(widget.credit_mining_button, Qt.LeftButton)
```

Comment:
Commented-out code, at this point I think you know what to do with it :)

Reply:
Again, that's part of the FIXME.

Reply:
Well, I think credit mining is something that should be fixed before this PR is merged. Maybe you can discuss this with @xoriole ?

Reply:
Yep, that's a good idea.